### PR TITLE
feat: configurable reaction TTL with permanent default

### LINKED ISSUE #34: [FEATURE] Configurable speech bubble TTL (reaction display duration)
## Problem

The speech bubble disappears after exactly 20 seconds. This is hardcoded in three places and not configurable. Some users want reactions to stay visible longer (or indefinitely until the next one arrives), others may want them shorter.

## How it works today

1. Buddy writes a reaction → `saveReaction()` stores it with `timestamp: Date.now()` in `reaction.$SID.json`
2. Status line / tmux popup reads the file every ~500ms
3. When `now - timestamp > 20s` → bubble clears → speech bubble disappears

### Hardcoded locations

| File | Line | Code | Context |
|---|---|---|---|
| `server/state.ts` | 236 | `Date.now() - data.timestamp > 20_000` | MCP server returns `null` when reaction >20s old |
| `statusline/buddy-status.sh` | 223 | `[ "$AGE" -lt 20 ] && FRESH=1` | Status line shows bubble only when <20s old |
| `popup/buddy-render.sh` | 26 | `REACTION_TTL=20` | tmux popup, used in `reaction_fresh()` |

## Proposed Solution

Add a `reactionTTL` field to `config.json` (default: 20 seconds), read it from all three locations instead of the hardcoded value.

- `bun run settings ttl <seconds>` to configure
- `0` = speech bubble stays visible until the next reaction replaces it
- Follows the same pattern as the existing `commentCooldown` setting

### Files to change

1. `server/state.ts` — add `reactionTTL` to `BuddyConfig`, use in `loadReaction()`
2. `statusline/buddy-status.sh` — read TTL from `config.json` instead of hardcoded `20`
3. `popup/buddy-render.sh` — read TTL from `config.json` instead of hardcoded `20`
4. `cli/settings.ts` — add `ttl` as a configurable setting

## Business Case

- Users who work in longer think/code cycles (5+ minutes between turns) never see the buddy react — the bubble is already gone by the time they look
- Users running demos or screenshots want the bubble to stay visible
- Power users who set `commentCooldown` to 0 (every reaction shows immediately) likely also want longer TTL so reactions don't flash by in 20 seconds
- Setting TTL to `0` (permanent until next reaction) makes the buddy feel more alive — there's always something in the speech bubble

### PR DESCRIPTION
**Pull request by 1270011**
Closes #34 — speech bubble duration is now configurable.

- **Default changed from 20s to 0** (permanent — bubble stays until next reaction), matching original Claude Code buddy behavior
- `bun run settings ttl <seconds>` to configure (0 = permanent, 1-300 = auto-hide after N seconds)
- Replaces hardcoded 20s TTL in all three locations (MCP server, status line, tmux popup)
- Removes unused `screenshot.png`
- builds on @Loky-Coffee #24 hooks/buddy-comment.sh: commentCooldown: 0 silently ignored

## Test plan

- [ ] `bun run settings` — shows TTL with default 0
- [ ] `bun run settings ttl 5` — set short TTL, trigger reaction, verify bubble disappears after 5s
- [ ] `bun run settings ttl 0` — verify bubble stays permanently
- [ ] Fresh install (no config.json) — verify default is 0 (permanent)
- [ ] Status line mode: bubble persists with TTL=0
- [ ] tmux popup mode: bubble persists with TTL=0 (not tested yet)

### CODE REVIEW

File: statusline/buddy-status.sh [hunk 1]
```diff
@@ -212,15 +212,22 @@ esac
 # ─── Reaction bubble (with TTL check) ────────────────────────────────────────
 BUBBLE=""
 REACTION_FILE="$HOME/.claude-buddy/reaction.$SID.json"
+REACTION_TTL=0
+CONFIG_FILE="$HOME/.claude-buddy/config.json"
+if [ -f "$CONFIG_FILE" ]; then
+    _ttl=$(jq -r '.reactionTTL // 0' "$CONFIG_FILE" 2>/dev/null || echo 20)
```
Comment (Loky-Coffee) [inline statusline/buddy-status.sh]: Small inconsistency here: the `jq` fallback uses `|| echo 20`, which produces `reactionTTL=20` if `jq` fails (e.g. on corrupted `config.json`). But the TypeScript default in `server/state.ts` is `reactionTTL: 0` (permanent).

So if `config.json` somehow becomes malformed, this path would silently fall back to a 20s TTL while the server-side `loadReaction()` would fall back to permanent. The two paths disagree on the "safe default".

Looks like a leftover from the previous hardcoded `20`. Suggested fix:

```diff
- _ttl=$(jq -r '.reactionTTL // 0' "$CONFIG_FILE" 2>/dev/null || echo 20)
+ _ttl=$(jq -r '.reactionTTL // 0' "$CONFIG_FILE" 2>/dev/null || echo 0)
```

Not a blocker — only matters on corrupted config — but worth aligning for consistency.

File: popup/buddy-render.sh [hunk 2]
```diff
@@ -37,6 +37,8 @@ if [ -f "$CONFIG_FILE" ]; then
   case "$_bp" in top|left) BUBBLE_POSITION="$_bp" ;; esac
   _sr=$(jq -r 'if .showRarity == false then "false" else "true" end' "$CONFIG_FILE" 2>/dev/null || echo "true")
   [ "$_sr" = "false" ] && SHOW_RARITY=0
+  _ttl=$(jq -r '.reactionTTL // 0' "$CONFIG_FILE" 2>/dev/null || echo 20)
```
Comment (Loky-Coffee) [inline popup/buddy-render.sh]: Same `|| echo 20` issue as in `statusline/buddy-status.sh`. Same suggested fix for consistency with the TypeScript `DEFAULT_CONFIG.reactionTTL = 0`:

```diff
- _ttl=$(jq -r '.reactionTTL // 0' "$CONFIG_FILE" 2>/dev/null || echo 20)
+ _ttl=$(jq -r '.reactionTTL // 0' "$CONFIG_FILE" 2>/dev/null || echo 0)
```

Looks like both files inherited the same copy-paste from the old hardcoded value. Two-line fix.

**Review by Loky-Coffee — COMMENTED** [review]
Really nice feature — solid implementation, backward compatible via the `{...DEFAULT_CONFIG, ...data}` merge in `loadConfig()`, and the default of `0` (permanent) matches the original buddy feel perfectly.

The two inline comments below are honestly just nitpicks — "complaining at a high level", as we say in German. Both are about the `|| echo 20` fallback in the shell scripts being inconsistent with the TypeScript default. It's only relevant if `jq` fails on a corrupted `config.json`, so it's a true edge case. Totally safe to merge as-is — I just wanted to flag the consistency while I had my eyes on it.

Also noticed the test plan marks tmux popup mode as "not tested yet". Logic looks identical to the status line path, so it should work, but a quick `tmux new && claude` verify before merge would be nice if you have a moment.

Thanks for shipping this, the permanent-bubble default is a really nice touch. 🐢

**Comment by 1270011** [comment]
> tmux new && claude

Thanks, `tmux new && claude` looks fine, tested now

**Comment by 1270011** [comment]
> Really nice feature — solid implementation, backward compatible via the `{...DEFAULT_CONFIG, ...data}` merge in `loadConfig()`, and the default of `0` (permanent) matches the original buddy feel perfectly.
> 
> The two inline comments below are honestly just nitpicks — "complaining at a high level", as we say in German. Both are about the `|| echo 20` fallback in the shell scripts being inconsistent with the TypeScript default. It's only relevant if `jq` fails on a corrupted `config.json`, so it's a true edge case. Totally safe to merge as-is — I just wanted to flag the consistency while I had my eyes on it.
> 
> Also noticed the test plan marks tmux popup mode as "not tested yet". Logic looks identical to the status line path, so it should work, but a quick `tmux new && claude` verify before merge would be nice if you have a moment.
> 
> Thanks for shipping this, the permanent-bubble default is a really nice touch. 🐢

All fixes & adjustments applied, thank you for review!
